connectors/src/connectors/slack/bot.ts

```diff
@@ -593,34 +593,48 @@ async function answerMessage(
   if (buildContentFragmentRes.isErr()) {
     return buildSlackMessageError(buildContentFragmentRes);
   }
+
   let conversation: ConversationType | undefined = undefined;
   let userMessage: UserMessageType | undefined = undefined;
+
   if (lastSlackChatBotMessage?.conversationId) {
-    if (buildContentFragmentRes.value) {
-      const contentFragmentRes = await dustAPI.postContentFragment({
+    // Check conversation existence (it might have been deleted between two messages).
+    const existsRes = await dustAPI.getConversation({
+      conversationId: lastSlackChatBotMessage.conversationId,
+    });
+
+    // If it doesn't exists, we will create a new one later.
+    if (existsRes.isOk()) {
+      if (buildContentFragmentRes.value) {
+        const contentFragmentRes = await dustAPI.postContentFragment({
+          conversationId: lastSlackChatBotMessage.conversationId,
+          contentFragment: buildContentFragmentRes.value,
+        });
+        if (contentFragmentRes.isErr()) {
+          return buildSlackMessageError(contentFragmentRes);
+        }
+      }
+
+      const messageRes = await dustAPI.postUserMessage({
         conversationId: lastSlackChatBotMessage.conversationId,
-        contentFragment: buildContentFragmentRes.value,
+        message: messageReqBody,
       });
-      if (contentFragmentRes.isErr()) {
-        return buildSlackMessageError(contentFragmentRes);
+      if (messageRes.isErr()) {
+        return buildSlackMessageError(messageRes);
       }
+      userMessage = messageRes.value;
+
+      const conversationRes = await dustAPI.getConversation({
+        conversationId: lastSlackChatBotMessage.conversationId,
+      });
+      if (conversationRes.isErr()) {
+        return buildSlackMessageError(conversationRes);
+      }
+      conversation = conversationRes.value;
     }
-    const messageRes = await dustAPI.postUserMessage({
-      conversationId: lastSlackChatBotMessage.conversationId,
-      message: messageReqBody,
-    });
-    if (messageRes.isErr()) {
-      return buildSlackMessageError(messageRes);
-    }
-    userMessage = messageRes.value;
-    const conversationRes = await dustAPI.getConversation({
-      conversationId: lastSlackChatBotMessage.conversationId,
-    });
-    if (conversationRes.isErr()) {
-      return buildSlackMessageError(conversationRes);
-    }
-    conversation = conversationRes.value;
-  } else {
+  }
+
+  if (!conversation || !userMessage) {
```

Comment:
The check on conversation should be enough, right?

Reply:
typescript complains :-)